### DISCUSSION TITLE
和 vuepress-plugin-right-anchor 一起使用会无法加载

### DISCUSSION THREAD

**Discussion by ahdg6** [vuepress-theme-hope/vuepress-theme-hope#1332] [Q&A] [answered]
Hope 主题自带的右侧标题锚点功能缺少配置，而插件 vuepress-plugin-right-anchor 的设计和可配置性又能更好地满足需求，但当这个插件和主题一起用时，页面会直接变为空白，有没有办法让主题和该插件共存？

谢谢。

**Comment by Mister-Hope** [comment]
你需要什么配置，欢迎详细说明

**Reply by ahdg6** [reply]
其实，那个插件很多功能我都觉得很有用，比如忽略指定标题，折叠标题锚点，自定义class，标题缩进，这些功能我觉得都有用，如果有可能，还是尽量一同使用吧，我也不好意思麻烦你实现那么多功能。

**Reply by ahdg6** [reply]
所以有没有可能一起用呢？那个插件确实已经很完善了，如果有办法的话记得踢我一脚，谢谢。

**Reply by Mister-Hope** [reply]
作为苦逼的理论物理研究生，我还没放假，等放假了再说。那个插件可能就是针对默认主题布局来的，所以没法在我的主题上生效。回头我看看，可以移植的会考虑移植一部分。

**Comment by ahdg6** [comment]
新年快乐！ 关于这个问题有什么进展吗?

**Reply by Mister-Hope** [reply]
Emm，我再详细的说一下 roadmap。我的寒假是先计划把主题和插件在v2下重写（这是一个两万多行代码，加文档配置文件等十万+的项目）。然后用svelte 重构 [@waline/client](https://github.com/walinejs/waline) 之后才会有时间研究你这个问题，至少还需要半个月来完成上述工作。

**Comment by Mister-Hope** [comment, marked as answer]
目前主题已经基本稳定，后续工作是会迁移一些主题在V1曾经使用的第三方插件。

目前主题使用了 `.theme-hope-content` 类，大概率没法兼容你说的插件，而且我也没看代码。如果你对主题的右侧侧边栏的可定制性有合理的新要求，欢迎提出，可酌情增加。

**Reply by ahdg6** [reply]
那个插件支持 v2 的，主要还是那个插件的设计对手机支持好，而且可以自定义折叠，你可以亲自去装一下

https://github.com/xuekai-china/vuepress-plugin-right-anchor

这是他们的演示
http://www.xwbuilders.com:9018/soonspacejs/Docs/1.x/api/basics/sbm.html

**Reply by Mister-Hope** [reply]
你自己试一下现在是否冲突，如果冲突的话可以自己找原因，就只要你有合理的兼容方案我这边可以配合你改。另外我看了一下，我对它的美观度表示质疑.....
